### PR TITLE
BUG: inconsistency in dtype of replace() #44864

### PR DESCRIPTION
**Pull request by vsydoriak99**
- [ ] closes #xxxx
- [x] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
can you write a test for this

**Comment by vsydoriak99** [comment]
Hello @jreback and @jbrockmendel, 
0cdfa4d1c1dfda5257e3838d06a145de9403a278 - has the same changes as in this last commit.
But 0cdfa4d1c1dfda5257e3838d06a145de9403a278  completed all tests succsesfull. 
Looks like that here is some problem with https://codecov.io, not in my code.

Could you please help me?

**Comment by jreback** [comment]
@vsydoriak99 this PR needs a test

**Comment by jbrockmendel** [comment]
@vsydoriak99 don't worry about the codecov issue, that usually resolves itself if we wait a while.

What this PR needs is a test for the fixed behavior.  In either tests.frame.methods.test_replace or tests.series.methods.test_replace

File: pandas/tests/series/methods/test_replace.py [hunk 1]
```diff
@@ -512,3 +512,10 @@ def test_pandas_replace_na(self):
         result = ser.replace(regex_mapping, regex=True)
         exp = pd.Series(["CC", "CC", "CC-REPL", "DD", "CC", "", pd.NA], dtype="string")
         tm.assert_series_equal(result, exp)
+
+    def test_replace_regex_dtype(self):
+        # GH-48644
+        s = pd.Series(['0'])
```
Comment (jbrockmendel) [inline pandas/tests/series/methods/test_replace.py:518]: nitpick: "ser" instead of "s"

Comment (jbrockmendel) [reply pandas/tests/series/methods/test_replace.py:518]: can you rebase, address this comment, and ping on green

**Comment by jreback** [comment]
@vsydoriak99 if you'd merge master and address comments

**Comment by jbrockmendel** [comment]
@vsydoriak99 can you rebase and address comments

**Comment by mroeschke** [comment]
Thanks for the PR, but it appears to have gone stale. If interested in continuing, please address the merge conflict and comments and we can reopen.

**Pull request closed without merging**